core/iwasm/interpreter/wasm_loader.c

```diff
@@ -3440,9 +3457,62 @@ load(const uint8 *buf, uint32 size, WASMModule *module, char *error_buf,
     return false;
 }
 
+#if (WASM_ENABLE_MULTI_MODULE != 0) && (WASM_ENABLE_LIBC_WASI != 0)
+/*
+ * refer to
+ * https://github.com/WebAssembly/WASI/blob/main/design/application-abi.md
+ */
+static bool
+check_wasi_abi_compatibility(const WASMModule *module, bool main_module,
+                             char *error_buf, uint32 error_buf_size)
+{
+    WASMExport *start = NULL, *initialize = NULL, *memory = NULL;
+
+    /*
+     * if a module doesn't depends on any library, it imports nothing.
+     * and its `is_wasi_module` flag is false. but still export below entries.
+     */
+
+    if (main_module) {
+        start = wasm_loader_find_export(module, "", "_start", EXPORT_KIND_FUNC,
+                                        error_buf, error_buf_size);
+        if (!start) {
+            /* without _start(), it maybe a reactor which is run via "iwasm -f
+             * XXX reactor" */
+            LOG_VERBOSE("A command should export _start function by default");
+        }
+    }
+    else {
+        initialize =
+            wasm_loader_find_export(module, "", "_initialize", EXPORT_KIND_FUNC,
+                                    error_buf, error_buf_size);
+        if (!initialize) {
+            set_error_buf(
```

Comment:
The document mentions that `A reactor may export a function named _initialize.. If an _initialize export is present`, my understanding is that `_initialize` function can be exported and also be unexported?

Reply:
A reactor always has an exported `_initialize`. Hasn't find any exception.

Reply:
OK. And should we also check that the sub-module should not export _start func, and main-module should not export _initialize func?

Reply:
Not sure about that. Clang should make sure that `_start` and `_initialize` can not be exported at the same time. But it is OK if we want to be safe first. I am open to both.

Reply:
Plus, if needs a check about signatures of `_start` and `_initialize`, we should have it done here.